src/sys/event.rs

```diff
@@ -3,280 +3,255 @@
+    kevent: libc::kevent,
 }
 
-#[cfg(not(any(target_os = "dragonfly", target_os = "netbsd")))]
-#[repr(i16)]
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub enum EventFilter {
-    EVFILT_READ = -1,
-    EVFILT_WRITE = -2,
-    EVFILT_AIO = -3,
-    EVFILT_VNODE = -4,
-    EVFILT_PROC = -5,
-    EVFILT_SIGNAL = -6,
-    EVFILT_TIMER = -7,
-    EVFILT_MACHPORT = -8,
-    EVFILT_FS = -9,
-    EVFILT_USER = -10,
-    // -11: unused
-    EVFILT_VM = -12,
-    EVFILT_SYSCOUNT = 13
-}
+#[cfg(any(target_os = "openbsd", target_os = "freebsd",
+          target_os = "dragonfly", target_os = "macos"))]
+type type_of_udata = *mut ::c_void;
+#[cfg(any(target_os = "netbsd"))]
+type type_of_udata = intptr_t;
 
-#[cfg(target_os = "dragonfly")]
-#[repr(i16)] // u_short
+#[cfg(not(target_os = "netbsd"))]
+type type_of_event_filter = i16;
+#[cfg(not(target_os = "netbsd"))]
+#[repr(i16)]
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum EventFilter {
-    EVFILT_READ = -1,
-    EVFILT_WRITE = -2,
-    EVFILT_AIO = -3,
-    EVFILT_VNODE = -4,
-    EVFILT_PROC = -5,
-    EVFILT_SIGNAL = -6,
-    EVFILT_TIMER = -7,
-    EVFILT_EXCEPT = -8,
-    EVFILT_USER = -9,
+    EVFILT_AIO = libc::EVFILT_AIO,
+    #[cfg(target_os = "dragonfly")]
+    EVFILT_EXCEPT = libc::EVFILT_EXCEPT,
+    #[cfg(any(target_os = "macos",
+              target_os = "dragonfly",
+              target_os = "freebsd"))]
+    EVFILT_FS = libc::EVFILT_FS,
+    #[cfg(target_os = "freebsd")]
+    EVFILT_LIO = libc::EVFILT_LIO,
+    #[cfg(target_os = "macos")]
+    EVFILT_MACHPORT = libc::EVFILT_MACHPORT,
+    EVFILT_PROC = libc::EVFILT_PROC,
+    EVFILT_READ = libc::EVFILT_READ,
+    EVFILT_SIGNAL = libc::EVFILT_SIGNAL,
+    EVFILT_TIMER = libc::EVFILT_TIMER,
+    #[cfg(any(target_os = "macos",
+              target_os = "dragonfly",
+              target_os = "freebsd"))]
+    EVFILT_USER = libc::EVFILT_USER,
+    #[cfg(target_os = "macos")]
+    EVFILT_VM = libc::EVFILT_VM,
+    EVFILT_VNODE = libc::EVFILT_VNODE,
+    EVFILT_WRITE = libc::EVFILT_WRITE,
 }
 
 #[cfg(target_os = "netbsd")]
+type type_of_event_filter = i32;
+#[cfg(target_os = "netbsd")]
 #[repr(u32)]
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum EventFilter {
-    EVFILT_READ = 0,
-    EVFILT_WRITE = 1,
-    EVFILT_AIO = 2,
-    EVFILT_VNODE = 3,
-    EVFILT_PROC = 4,
-    EVFILT_SIGNAL = 5,
-    EVFILT_TIMER = 6,
-    EVFILT_SYSCOUNT = 7
+    EVFILT_READ = libc::EVFILT_READ,
+    EVFILT_WRITE = libc::EVFILT_WRITE,
+    EVFILT_AIO = libc::EVFILT_AIO,
+    EVFILT_VNODE = libc::EVFILT_VNODE,
+    EVFILT_PROC = libc::EVFILT_PROC,
+    EVFILT_SIGNAL = libc::EVFILT_SIGNAL,
+    EVFILT_TIMER = libc::EVFILT_TIMER,
 }
 
-#[cfg(not(any(target_os = "dragonfly", target_os = "netbsd")))]
+#[cfg(any(target_os = "macos",
+          target_os = "freebsd",
+          target_os = "dragonfly"))]
+type type_of_event_flag = u16;
+#[cfg(any(target_os = "macos",
+          target_os = "freebsd",
+          target_os = "dragonfly"))]
 bitflags!(
```

Comment:
I think you can now use the `libc_bitflags` macro, that allows to reduce repetition. You can find examples elsewhere in nix.